File: exercises/D305/solution.pas
var
    s: string;
    len, sum: longint;
begin
    readln(s);
    len := length(s);
    if (len = 8) then
        sum := 9 * (ord(s[1]) - 55) + 8 * (ord(s[2]) - 55)
    else
        sum := 9 * 36 + 8 * (ord(s[1]) - 55);
    sum := sum + 7 * (ord(s[len - 5]) - 48);
    sum := sum + 6 * (ord(s[len - 4]) - 48);
    sum := sum + 5 * (ord(s[len - 3]) - 48);
    sum := sum + 4 * (ord(s[len - 2]) - 48);
    sum := sum + 3 * (ord(s[len - 1]) - 48);
    sum := sum + 2 * (ord(s[len]) - 48);
    write(s);
    if (sum mod 11 = 0) then
        writeln('(0)')
    else if (sum mod 11 = 1) then
        writeln('(A)')
    else
        writeln('(', 11 - sum mod 11, ')')
end.

File: exercises/D305/solution.py
s = input()
length = len(s)
if length == 8:
    sum = 9 * (ord(s[0]) - ord('A') + 10) + 8 * (ord(s[1]) - ord('A') + 10)
else:
    sum = 9 * 36 + 8 * (ord(s[0]) - ord('A') + 10)
sum += 7 * (ord(s[length - 6]) - ord('0'))
sum += 6 * (ord(s[length - 5]) - ord('0'))
sum += 5 * (ord(s[length - 4]) - ord('0'))
sum += 4 * (ord(s[length - 3]) - ord('0'))
sum += 3 * (ord(s[length - 2]) - ord('0'))
sum += 2 * (ord(s[length - 1]) - ord('0'))
if sum % 11 == 0:
    print(s + '(0)')
elif sum % 11 == 1:
    print(s + '(A)')
else:
    print(s + '(' + str(11 - sum % 11) + ')')
